src/indexer/segment_writer.rs

```diff
@@ -315,8 +330,15 @@ impl SegmentWriter {
     }
 }
 
-// This method is used as a trick to workaround the borrow checker
-fn write(
+/// This method is used as a trick to workaround the borrow checker
+/// Writes a view of a segment by pushing information
+/// to the `SegmentSerializer`.
+///
+/// `doc_id_map` is used to map to the new doc_id order.
+///
+/// # Returns
+/// The number of documents in the segment.
```

Comment:
Probably a copy paste from the merger. This method doesn't return anything.

Reply:
Thank you for the catch!